tests/e2e_test.go

```diff
@@ -0,0 +1,555 @@
+		receivedWarpMessage              *avalancheWarp.Message
+		chainBWSClient                   ethclient.Client
+		chainARPCClient, chainBRPCClient ethclient.Client
+		chainARPCURI, chainBRPCURI       string
+		chainAIDInt                      *big.Int
+		chainBIDInt                      *big.Int
+		payload                          []byte
+		relayerCmd                       *exec.Cmd
+		relayerCancel                    context.CancelFunc
+	)
+
+	fundedKey, err = crypto.HexToECDSA(fundedKeyStr)
+	if err != nil {
+		panic(err)
+	}
+
+	ginkgo.It("Setup subnet URIs", ginkgo.Label("Relayer", "Setup"), func() {
+		subnetIDs = manager.GetSubnets()
+		Expect(len(subnetIDs)).Should(Equal(2))
+
+		subnetA = subnetIDs[0]
+		subnetADetails, ok := manager.GetSubnet(subnetA)
+		Expect(ok).Should(BeTrue())
+		Expect(len(subnetADetails.ValidatorURIs)).Should(Equal(5))
+		blockchainIDA = subnetADetails.BlockchainID
+		chainANodeURIs = append(chainANodeURIs, subnetADetails.ValidatorURIs...)
+
+		subnetB = subnetIDs[1]
+		subnetBDetails, ok := manager.GetSubnet(subnetB)
+		Expect(ok).Should(BeTrue())
+		Expect(len(subnetBDetails.ValidatorURIs)).Should(Equal(5))
+		blockchainIDB = subnetBDetails.BlockchainID
+		chainBNodeURIs = append(chainBNodeURIs, subnetBDetails.ValidatorURIs...)
+
+		log.Info("Created URIs for both subnets", "ChainAURIs", chainANodeURIs, "ChainBURIs", chainBNodeURIs, "blockchainIDA", blockchainIDA, "blockchainIDB", blockchainIDB)
+
+		chainAWSURI := httpToWebsocketURI(chainANodeURIs[0], blockchainIDA.String())
+		chainARPCURI = httpToRPCURI(chainANodeURIs[0], blockchainIDA.String())
+		log.Info("Creating ethclient for blockchainA", "wsURI", chainAWSURI, "rpcURL, chainARPCURI")
+		chainARPCClient, err = ethclient.Dial(chainARPCURI)
+		Expect(err).Should(BeNil())
+
+		chainAIDInt, err = chainARPCClient.ChainID(context.Background())
+		Expect(err).Should(BeNil())
+
+		chainBWSURI := httpToWebsocketURI(chainBNodeURIs[0], blockchainIDB.String())
+		chainBRPCURI = httpToRPCURI(chainBNodeURIs[0], blockchainIDB.String())
+		log.Info("Creating ethclient for blockchainB", "wsURI", chainBWSURI)
+		chainBWSClient, err = ethclient.Dial(chainBWSURI)
+		Expect(err).Should(BeNil())
+		chainBRPCClient, err = ethclient.Dial(chainBRPCURI)
+		Expect(err).Should(BeNil())
+
+		chainBIDInt, err = chainBRPCClient.ChainID(context.Background())
+		Expect(err).Should(BeNil())
+
+		log.Info("Finished setting up e2e test subnet variables")
+	})
+
+	ginkgo.It("Deploy Teleporter Contract", ginkgo.Label("Relayer", "Deploy Teleporter"), func() {
+		ctx := context.Background()
+
+		// Read in the Teleporter contract information
+		teleporterContractAddress = common.BytesToAddress(readHexTextFile("./tests/UniversalTeleporterMessengerContractAddress.txt"))
+		teleporterDeployerAddress := common.BytesToAddress(readHexTextFile("./tests/UniversalTeleporterDeployerAddress.txt"))
+		teleporterDeployerTransaction := readHexTextFile("./tests/UniversalTeleporterDeployerTransaction.txt")
```

Comment:
It looks like we read the hex string, convert it to a `[]byte`, but then the only place we use it we first convert it back to a hex string. Could we just read the string from the file and use it directly?

Reply:
We could, but if we wanted to still use `readHexTextFile`, we'd have to alter it to return `string`, then convert that `string` to `common.Address` for the other two vars here. IMO, the current implementation is more readable in the test file, even though there are unnecessary conversions as you pointed out.

Reply:
You can use `common.HexToAddress` to get the same readability if you want to change `readHexTextFile` to return a string.